File: NotesServer/Models/NoteImage.cs
using System;
using SQLite;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace NotesServer.Models
{
    public class NoteImage : DatabaseEntity
    {
        [PrimaryKey, AutoIncrement]
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [Required]
        [JsonPropertyName("noteId")]
        public int NoteId { get; set; }

        [Required]
        [JsonPropertyName("fileName")]
        public string FileName { get; set; } = string.Empty;

        [Required]
        [JsonPropertyName("imageData")]
        public byte[] ImageData { get; set; } = Array.Empty<byte>();

        public DateTime CreatedAt { get; set; }
    }
} 

File: NotesServerGO/models/note_image.go
package models

import "time"

// NoteImage представляет изображение, прикрепленное к заметке.
type NoteImage struct {
	Id         int64     `json:"id,omitempty" db:"Id"`                  // Может отсутствовать для новых изображений в бэкапе
	NoteId     int64     `json:"note_id" db:"NoteId"`                   // ID заметки, к которой относится изображение
	ImagePath  string    `json:"-" db:"ImagePath"`                      // Путь к файлу на сервере (не приходит от клиента в бэкапе)
	FileName   string    `json:"file_name" db:"FileName"`               // Имя файла, которое клиент присылает и ожидает
	ImageData  string    `json:"image_data,omitempty" db:"-"`           // Base64 строка данных изображения (от клиента), не хранится в БД
	DatabaseId int64     `json:"database_id,omitempty" db:"DatabaseId"` // ID совместной БД, к которой относится NoteId
	CreatedAt  time.Time `json:"-" db:"CreatedAt"`                      // ИСПРАВЛЕНИЕ: убираю из JSON
	UpdatedAt  time.Time `json:"-" db:"UpdatedAt"`                      // ИСПРАВЛЕНИЕ: убираю из JSON
}
